storm-server/src/main/java/org/apache/storm/container/oci/RuncLibContainerManager.java

```diff
@@ -0,0 +1,631 @@
+import org.apache.storm.utils.ServerUtils;
+import org.apache.storm.utils.Utils;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+public class RuncLibContainerManager extends OciContainerManager {
+    private static final Logger LOG = LoggerFactory.getLogger(RuncLibContainerManager.class);
+
+    private OciImageTagToManifestPluginInterface imageTagToManifestPlugin;
+    private OciManifestToResourcesPluginInterface manifestToResourcesPlugin;
+    private OciResourcesLocalizerInterface ociResourcesLocalizer;
+    private ObjectMapper mapper;
+    private int layersToKeep;
+    private String seccomp;
+
+    private static final String RESOLV_CONF = "/etc/resolv.conf";
+    private static final String HOSTNAME = "/etc/hostname";
+    private static final String HOSTS = "/etc/hosts";
+    private static final String OCI_CONFIG_JSON = "oci-config.json";
+
+    private static final String SQUASHFS_MEDIA_TYPE = "application/vnd.squashfs";
+
+    //CPU CFS (Completely Fair Scheduler) period
+    private static final long CPU_CFS_PERIOD_US = 100000;
+
+    private Map<String, Long> workerToContainerPid = new ConcurrentHashMap<>();
+    private Map<String, ExitCodeCallback> workerToExitCallback = new ConcurrentHashMap<>();
+    private Map<String, String> workerToUser = new ConcurrentHashMap<>();
+    private StormTimer checkContainerAliveTimer;
+
+    @Override
+    public void prepare(Map<String, Object> conf) throws IOException {
+        super.prepare(conf);
+
+        imageTagToManifestPlugin = chooseImageTagToManifestPlugin();
+        imageTagToManifestPlugin.init(conf);
+
+        manifestToResourcesPlugin = chooseManifestToResourcesPlugin();
+        manifestToResourcesPlugin.init(conf);
+
+        ociResourcesLocalizer = chooseOciResourcesLocalizer();
+        ociResourcesLocalizer.init(conf);
+
+        layersToKeep = ObjectReader.getInt(
+                conf.get(DaemonConfig.STORM_OCI_LAYER_MOUNTS_TO_KEEP),
+                100
+        );
+
+        mapper = new ObjectMapper();
+
+        if (seccompJsonFile != null) {
+            seccomp = new String(Files.readAllBytes(Paths.get(seccompJsonFile)));
+        }
+
+        if (checkContainerAliveTimer == null) {
+            checkContainerAliveTimer =
+                new StormTimer("CheckRuncContainerAlive", Utils.createDefaultUncaughtExceptionHandler());
+            checkContainerAliveTimer
+                .scheduleRecurring(0, (Integer) conf.get(DaemonConfig.SUPERVISOR_MONITOR_FREQUENCY_SECS), () -> {
+                    try {
+                        checkContainersAlive();
+                    } catch (Exception e) {
+                        //Ignore
+                        LOG.debug("The CheckRuncContainerAlive thread has exception. Ignored", e);
```

Comment:
Is it expected that checkContainersAlive() throw an exception without a message?

Reply:
changing to warn